src/node_sea.cc

```diff
@@ -238,6 +278,49 @@ std::optional<SeaConfig> ParseSingleExecutableConfig(
   return result;
 }
 
+std::optional<std::string> GenerateCodeCache(std::string_view main_path,
+                                             std::string_view main_script) {
+  RAIIIsolate raii_isolate;
+  Isolate* isolate = raii_isolate.get();
+
+  HandleScope handle_scope(isolate);
+  Local<Context> context = Context::New(isolate);
+  Context::Scope context_scope(context);
+
+  errors::PrinterTryCatch bootstrapCatch(
+      isolate, errors::PrinterTryCatch::kPrintSourceLine);
+
+  Local<String> filename;
+  if (!String::NewFromUtf8(isolate, main_path.data()).ToLocal(&filename)) {
+    return {};
+  }
+
+  Local<String> content;
+  if (!String::NewFromUtf8(isolate, main_script.data()).ToLocal(&content)) {
+    return {};
+  }
```

Comment:
Technically it is possible, but in this particular code: What scenario does String::NewFromUtf8 fail for `std::string_view`?

Reply:
I think these are the cases:
- when the length exceeds the max limit: https://github.com/nodejs/node/blob/be469d85da927b27b5b35f6e23f3e573309e2ae1/deps/v8/src/api/api.cc#L7505-L7506
- a number of cases in this function which I haven't deeply looked into: https://github.com/nodejs/node/blob/be469d85da927b27b5b35f6e23f3e573309e2ae1/deps/v8/src/heap/factory.cc#L692-L734